addons/account/wizard/account_move_send.py

```diff
@@ -411,7 +411,9 @@ def _hook_if_success(self, moves_data, from_cron=False, allow_fallback_pdf=False
 
         # Send mail.
         if send_mail:
-            self._send_mails(moves_data)
+            mail_moves_data = {move: move_data for move, move_data in moves_data.items() if move.partner_id.email}
+            if mail_moves_data:
+                self._send_mails(mail_moves_data)
```

Comment:
Why don't we put this part of code directly in the `_send_mails`?

Reply:
Good idea, I have had a filter in the loop.